### ISSUE TITLE
Announcement: Resuming Maintenance and Upgrading Py-Googletrans Planned (v4.0)

### ISSUE THREAD

**Issue by ssut** [ssut/py-googletrans#411]
First of all, I apologize for not maintaining this project for quite a while. My life and work have been so busy, and I just didn't have the enough time to maintain this project properly. Yes, some people may know I was working on other (OSS) projects, but that doesn't mean I abandoned this project. I just thought py-googletrans no longer needed frequent updates because, you guys know, there is already a *mature and official* Google Translate API.

Yes, here's the good news for you guys: I'm back and ready to re-dive into this project. Here's what I'm planning for now: (1) fixing stuffs (2) modernizing the entire code (yeah, the Python ecosystem has changed and grew a lot).

Thanks so much to everyone who reported issues and using this project even during such downtime. I really appreciate it.

I plan to resolve all the things up to 3 months as I have some planned time off (holiday) things. However there might be some unexpected schedule changes, but I'll do my best to release even an alpha/beta version ASAP.

Thanks so much again to everyone who waited for this project to be re-maintained and taken further.

---

If you guys have experienced any issues, please report them on this issue.

**Comment by Josanshuo** [comment]
Thank you for resuming the work. 

Wondering if the `httpx` version could be loosen, so other packages which uses newer version of `httpx` could be resolved by pip?

For example, I'm trying to upgrade the https://github.com/gradio-app/gradio while using googletrans, it requires the `httpx>=0.24.1` which pip couldn't resolve it as googletrans require the `httpx==0.13.3`.

https://github.com/gradio-app/gradio/blob/b635a317142836c92a0eb20693ad2d4ee43c61f6/requirements.txt#L6

**Comment by ssut** [comment]
> Thank you for resuming the work.
> 
> Wondering if the `httpx` version could be loosen, so other packages which uses newer version of `httpx` could be resolved by pip?
> 
> For example, I'm trying to upgrade the https://github.com/gradio-app/gradio while using googletrans, it requires the `httpx>=0.24.1` which pip couldn't resolve it as googletrans require the `httpx==0.13.3`.
> 
> https://github.com/gradio-app/gradio/blob/b635a317142836c92a0eb20693ad2d4ee43c61f6/requirements.txt#L6

Yes, I’m planning to loosen the version constraint of httpx in a future update. While I’m currently rewriting the entire things, the version constraint will be addressed as well.

**Comment by jonbesga** [comment]
Hi, is there any update on the matter of httpx? I'm also facing the version constraint

**Comment by c-goosen** [comment]
I've done some of the work to have3 a more open httpx version if you want to rather just merge in some changes than do a lot of work yourself. https://github.com/c-goosen/py-googletrans
Could do a PR

**Comment by c-goosen** [comment]
https://github.com/ssut/py-googletrans/pull/416

**Comment by xdevfaheem** [comment]
@ssut any updates regarding http-core dependency conflict?, because of that i couldn't connect it on gradio-app. 

luckily i cloned and installed @c-goosen fork, fortunately, it fixed the conflict. can't thank you enough @c-goosen !

**Comment by soorower** [comment]
@ssut  Please, add proxy support.

**Comment by ssut** [comment]
I’ve opened a new pull request to get ready for the v4.0.0 (next major) release. It fixes most reported issues, modernizes the code, switches to async-only APIs, and adds complete typing. If anyone's planning to send a PR, please set the target branch to this. Thanks a lot for waiting!

**Comment by jonbesga** [comment]
Awesome! Thank you for your work!

**Comment by ssut** [comment]
Version 4 of py-googletrans is now available. Thank you all for waiting for this release. If you encounter any issues, please report them at any time. I am now going to close this issue.
